docs/ingestion/native-batch.md

```diff
@@ -1368,6 +1368,48 @@ Compared to the other native batch InputSources, SQL InputSource behaves differe
 * Similar to file-based input formats, any updates to existing data will replace the data in segments specific to the intervals specified in the `granularitySpec`.
 
 
+### Combining Input Source
+
+The Combining input source is used to read data from multiple InputSources. This input source should be only used if all the delegate input sources are
+ _splittable_ and can be used by the [Parallel task](#parallel-task). This input source will identify the splits from its delegates and each split will be processed by a worker task. Similar to other input sources, this input source supports a single `inputFormat`. Therefore, please note that delegate input sources requiring an `inputFormat` must have the same format for input data.
```

Comment:
> please note that delegate input sources requiring an `inputFormat` must have the same format for input data.

Hmm, thinking about this requirement, could it be more useful if you can read multiple inputSources even when they are formatted differently?

Reply:
Yeah did think about that and it would be definitely be more useful if we can handle multiple inputFormats as well. For doing this, we might have to push `inputFormat` to be a part of the `inputSource` config. Since it would impact other inputSources as well, I thought it would be better to investigate that separately and have an independent PR for that.

Reply:
Sounds good to me. When we add support for multiple inputFormats, we should probably think about how we can make batch and streaming ingestion spec less diverse in describing inputSource and inputFormat as streaming ingestion currently doesn't require inputSource.